rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -434,9 +443,17 @@ def _decode_transaction(
         # Check if any rules should run due to the 4bytes signature of the input data
         fourbytes = transaction.input_data[:4]
         input_data_rules = self.rules.input_data_rules.get(fourbytes)
+        monerium_special_handling_event = False
 
         # decode transaction logs from the receipt
         for tx_log in tx_receipt.logs:
+            if (
```

Comment:
why not just move this to line 513 and not have an extra if for nothing?

Reply:
This is here to avoid an extra loop if it isn't needed. What it does is checking if there is a monerium event and if that is the case set the `monerium_special_handling_event` flag. In case we have this flag set to true then later we iterate over all the events to find those that need to be updated. The condition is here to avoid looping twice all the events/logs